ouroboros-network/src/Ouroboros/Network/PeerSelection/Simple.hs

```diff
@@ -32,43 +34,44 @@ import           Ouroboros.Network.PeerSelection.RootPeersDNS
 withPeerSelectionActions
   :: Tracer IO TraceLocalRootPeers
   -> Tracer IO TracePublicRootPeers
+  -> TimeoutFn IO
   -> PeerSelectionTargets
-  -> Map Socket.SockAddr PeerAdvertise
+  -> [(Int, Map RelayAddress PeerAdvertise)]
   -- ^ static local root peers
   -> [(DomainAddress, PeerAdvertise)]
   -- ^ local root peers
   -> [DomainAddress]
   -- ^ public root peers
   -> PeerStateActions Socket.SockAddr peerconn IO
-  -> (Async IO () -> PeerSelectionActions Socket.SockAddr peerconn IO -> IO a)
-  -- ^ continuation, which allows to tra
+  -> (Maybe (Async IO Void) -> PeerSelectionActions Socket.SockAddr peerconn IO -> IO a)
+  -- ^ continuation, recieves a handle to the local roots peer provider thread
+  -- (only if local root peers where non-empty).
   -> IO a
-withPeerSelectionActions localRootTracer publicRootTracer targets staticLocalRootPeers localRootPeers publicRootPeers peerStateActions k = do
-    localRootsVar <- newTVarIO mempty
-    withTimeoutSerial $ \timeout ->
-      withAsync
-        (localRootPeersProvider
-          localRootTracer
-          timeout
-          DNS.defaultResolvConf
-          localRootsVar
-          localRootPeers)
-        $ \thread ->
-          k thread
-            PeerSelectionActions {
-                readPeerSelectionTargets = pure targets,
-                readLocalRootPeers = do
-                  localRoots <- readTVar localRootsVar
-                  pure [(0, foldr Map.union staticLocalRootPeers localRoots)],
-                requestPublicRootPeers = requestPublicRootPeers timeout,
-                requestPeerGossip = \_ -> pure [],
-                peerStateActions
-              }
+withPeerSelectionActions localRootTracer publicRootTracer timeout targets _staticLocalRootPeers localRootPeers publicRootPeers peerStateActions k = do
+    localRootsVar <- newTVarIO Map.empty
+    let peerSelectionActions = PeerSelectionActions {
+            readPeerSelectionTargets = pure targets,
+            readLocalRootPeers = pure [],
+            requestPublicRootPeers,
+            requestPeerGossip = \_ -> pure [],
+            peerStateActions
+          }
+    case localRootPeers of
+      []       -> k Nothing peerSelectionActions
+      (a : as) ->
+        withAsync
```

Comment:
We don't need to run `localRootPeersProvider` at this stage, this should be introduced at a later commit.

Reply:
This commit can be smashed with the previous one.

Reply:
done